app/src/main/java/com/gazlaws/codeboard/CodeBoardIME.java

```diff
@@ -272,8 +274,19 @@ public void onKey(int primaryCode, int[] KeyCodes) {
 //                        }
                 }
                 if (ke != 0) {
+
                     Log.i(getClass().getSimpleName(), "onKey: keyEvent " + ke);
-                    ic.sendKeyEvent(new KeyEvent(0, 0, KeyEvent.ACTION_DOWN, ke, 0, meta));
+
+                    /*
+                     *   The if statement was added in order to prevent the space button
+                     *   from having an action down event attached to it.
+                     *   Reason being that we first want to check
+                     *   whether the space button has been long pressed or not
+                     *   and afterwards produce the right output to the screen.
+                     *   Note : The ACTION_UP KeyEvent doesn't bother us either way.
```

Comment:
Can you change this to "TODO: Investigate whether KeyEvent.ACTION_UP is still required".

When testing on my phone, preventing both actions worked, so I suspect that it's an emulator issue. But we don't need to fix it right now.

Reply:
Okay, but just to inform you, I managed to run it on an Android device and it worked either way! Meaning, even when I had the ACTION_UP KeyEvent enabled  it worked, but it also worked, when I didn't had it. To sum up, the only way it doesn't work is when I had the ACTION_DOWN KeyEvent enabled. If the ACTION_DOWN KeyEvent is disabled it always works.

Reply:
For me, all scenarios produce the same results as the Android device when I test them on the emulator.

Reply:
> For me, all scenarios produce the same results as the Android device when I test them on the emulator.

I'll add the TODO because it is strange it doesn't work for you as well.